service/src/main/java/bio/terra/pipelines/dependencies/leonardo/AppUtils.java

```diff
@@ -0,0 +1,199 @@
+package bio.terra.pipelines.dependencies.leonardo;
+
+import bio.terra.pipelines.app.configuration.external.LeonardoServerConfiguration;
+import bio.terra.pipelines.app.configuration.internal.ImputationConfiguration;
+import java.time.OffsetDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+import org.broadinstitute.dsde.workbench.client.leonardo.model.AppStatus;
+import org.broadinstitute.dsde.workbench.client.leonardo.model.AppType;
+import org.broadinstitute.dsde.workbench.client.leonardo.model.ListAppResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AppUtils {
+
+  private final LeonardoServerConfiguration leonardoServerConfiguration;
+  private final ImputationConfiguration imputationConfiguration;
+
+  private static final Logger logger = LoggerFactory.getLogger(AppUtils.class);
+
+  public AppUtils(
+      LeonardoServerConfiguration leonardoServerConfiguration,
+      ImputationConfiguration imputationConfiguration) {
+    this.leonardoServerConfiguration = leonardoServerConfiguration;
+    this.imputationConfiguration = imputationConfiguration;
+  }
+
+  int appComparisonFunction(ListAppResponse a, ListAppResponse b, List<AppType> appTypeList) {
```

Comment:
so I know this is just pulled from CBAS but it would really improve readability for me to use more meaningful variable names than `a` and `b` throughout this method and in suitableApps below